undirected.go

```diff
@@ -201,41 +188,25 @@ func (u *undirected[K, T]) edgesAreEqual(a, b Edge[T]) bool {
 	return false
 }
 
-func (u *undirected[K, T]) addEdge(sourceHash, targetHash K, edge Edge[T]) {
-	if _, ok := u.outEdges[sourceHash]; !ok {
-		u.outEdges[sourceHash] = make(map[K]Edge[T])
-	}
-	if _, ok := u.outEdges[targetHash]; !ok {
-		u.outEdges[targetHash] = make(map[K]Edge[T])
+func (u *undirected[K, T]) addEdge(sourceHash, targetHash K, edge Edge[K]) error {
+	err := u.store.AddEdge(sourceHash, targetHash, edge)
+	if err != nil {
+		return err
 	}
 
-	u.outEdges[sourceHash][targetHash] = edge
-	u.outEdges[targetHash][sourceHash] = edge
-
-	if _, ok := u.inEdges[targetHash]; !ok {
-		u.inEdges[targetHash] = make(map[K]Edge[T])
-	}
-	if _, ok := u.inEdges[sourceHash]; !ok {
-		u.inEdges[sourceHash] = make(map[K]Edge[T])
-	}
-
-	u.inEdges[targetHash][sourceHash] = edge
-	u.inEdges[sourceHash][targetHash] = edge
-}
-
-func (u *undirected[K, T]) adjacencies(vertexHash K) []K {
-	var adjacencyHashes []K
-
-	// An undirected graph creates an undirected edge as two directed edges in the opposite
-	// direction, so both the in-edges and the out-edges work here.
-	inEdges, ok := u.inEdges[vertexHash]
-	if !ok {
-		return adjacencyHashes
+	rEdge := Edge[K]{
+		Source: edge.Target,
+		Target: edge.Source,
```

Comment:
Looks like you've revealed a bug in the current version here because the unmodified `edge` instance is used for both "directions". I really need to investigate how this is not detected by the tests.